src/AppSystem/Launcher.vala

```diff
@@ -385,9 +413,31 @@ public class Dock.Launcher : BaseItem {
         }
     }
 
-    private void update_badge_revealer () {
-        badge_revealer.reveal_child = app.count_visible
-            && (notify_settings == null || !notify_settings.get_boolean ("do-not-disturb"));
+    private void update_badge_revealed () {
+        badge_fade.skip ();
+        badge_scale.skip ();
+
+        // Avoid a stutter at the beginning
+        badge.opacity = 0;
+
+        if (app.count_visible && (notify_settings == null || !notify_settings.get_boolean ("do-not-disturb"))) {
```

Comment:
What is the idea behind the DND check here? It looks like it skips this and will use the default up above and still animate. And then if it is removing the badge and DND is on, it will use the default but won't be reversed. Am I reading this right?

Reply:
Oooooh I see